File: frontend/lib/models/progress_history.dart
class ProgressHistory {
  final int? id;
  final String userId;
  final DateTime date;
  final int steps;
  final int pushUps;
  final int squats;
  final int plankSeconds;
  final int waterMl;

  ProgressHistory({
    this.id,
    required this.userId,
    required this.date,
    this.steps = 0,
    this.pushUps = 0,
    this.squats = 0,
    this.plankSeconds = 0,
    this.waterMl = 0,
  });

  factory ProgressHistory.fromJson(Map<String, dynamic> json) {
    return ProgressHistory(
      id: json['id'],
      userId: json['user_id'] ?? '',
      date: DateTime.parse(json['date']),
      steps: json['steps'] ?? 0,
      pushUps: json['push_ups'] ?? 0,
      squats: json['squats'] ?? 0,
      plankSeconds: json['plank_seconds'] ?? 0,
      waterMl: json['water_ml'] ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'user_id': userId,
      'date': date.toIso8601String().split('T')[0],
      'steps': steps,
      'push_ups': pushUps,
      'squats': squats,
      'plank_seconds': plankSeconds,
      'water_ml': waterMl,
    };
  }
}

File: backend/src/main/java/com/sempermove/backend/model/ProgressHistory.java
package com.sempermove.backend.model;

import jakarta.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "progress_history")
public class ProgressHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long userId;
    private LocalDate date = LocalDate.now();

    private int steps;
    private int pushUps;
    private int squats;
    private int plankSeconds;
    private int waterMl;

    public ProgressHistory() {}

    public ProgressHistory(Long userId, int steps, int pushUps, int squats, int plankSeconds, int waterMl) {
        this.userId = userId;
        this.steps = steps;
        this.pushUps = pushUps;
        this.squats = squats;
        this.plankSeconds = plankSeconds;
        this.waterMl = waterMl;
    }

    // --- getters & setters ---
    public Long getId() {
        return id;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    public int getSteps() {
        return steps;
    }

    public void setSteps(int steps) {
        this.steps = steps;
    }

    public int getPushUps() {
        return pushUps;
    }

    public void setPushUps(int pushUps) {
        this.pushUps = pushUps;
    }

    public int getSquats() {
        return squats;
    }

    public void setSquats(int squats) {
        this.squats = squats;
    }

    public int getPlankSeconds() {
        return plankSeconds;
    }

    public void setPlankSeconds(int plankSeconds) {
        this.plankSeconds = plankSeconds;
    }

    public int getWaterMl() {
        return waterMl;
    }

    public void setWaterMl(int waterMl) {
        this.waterMl = waterMl;
    }
}
